lib/semantics/expression.cc

```diff
@@ -134,6 +134,7 @@ class ArgumentAnalyzer {
   ArgumentAnalyzer(ExpressionAnalyzer &context, parser::CharBlock source)
     : context_{context}, source_{source} {}
   bool fatalErrors() const { return fatalErrors_; }
+  void set_allowAssumedType(bool x = true) { allowAssumedType_ = x; }
```

Comment:
This flag appears to be only set after construction in the two cases of function and subroutine references, and never cleared, so it could be specified with an additional Boolean argument to the second constructor, and also made a `const` data member.

Reply:
OK, good idea.